### PR TITLE
IGNITE-12773 : Reduce number of cluster deactivation methods in internal API.

### PR DESCRIPTION
**Pull request by Vladsz83**
IGNITE-12773 : Reduce number of cluster deactivation methods in internal API.

### CODE REVIEW

**Comment by Vladsz83** [comment]
{code:java}
1.	Removed
GridClientClusterState#active()

2.	Removed
GridClientClusterState#active(boolean active)

3.	Removed
IGridClusterStateProcessor#changeGlobalState(
    boolean activate,
    Collection<? extends BaselineNode> baselineNodes,
    boolean forceChangeBaselineTopology
)

4.	Removed
GridClusterStateProcessor#changeGlobalState(
    final boolean activate,
    Collection<? extends BaselineNode> baselineNodes,
    boolean forceChangeBaselineTopology,
    boolean isAutoAdjust
)

5.	Removed
GridClusterStateProcessor#changeGlobalState(
    final boolean activate,
    Collection<? extends BaselineNode> baselineNodes,
    boolean forceChangeBaselineTopology
)

6.	Remove 
GridClusterStateProcessor#changeGlobalState(
    ClusterState state,
    Collection<? extends BaselineNode> baselineNodes,
    boolean forceChangeBaselineTopology
)

7.	Added boolean isAutoAdjust to 
IGridClusterStateProcessor#changeGlobalState(
    ClusterState state,
    Collection<? extends BaselineNode> baselineNodes,
    boolean forceChangeBaselineTopology,
   /* here */ boolean isAutoAdjust /* here */
)

8.	Added @Override to 
/* here */ @Override /* here */
IGridClusterStateProcessor#changeGlobalState(
        ClusterState state,
        Collection<? extends BaselineNode> baselineNodes,
        boolean forceChangeBaselineTopology,
        boolean isAutoAdjust
    )

9.      Removed, combined with #8:
IGridClusterStateProcessor#changeGlobalState0(
        ClusterState state,
        BaselineTopology blt,
        boolean forceChangeBaselineTopology,
        boolean isAutoAdjust
    ) 
{code}

- [ ]

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java [hunk 1]
```diff
@@ -87,23 +89,23 @@
...
         IgniteEx ig1 = grid(0);
         IgniteEx ig2 = grid(1);
 
         assertFalse(ig1.active());
         assertFalse(ig2.active());
-        assertFalse(state.active());
+        assertEquals(state.state(), INACTIVE);
 
-        state.active(true);
+        state.state(ACTIVE, true);
```
Comment (NSAmelchev) [inline modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: There is no cluster deactivation, `forceDeactivation` should be false.

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/MvccJdbcTransactionFinishOnDeactivatedClusterSelfTest.java [hunk 2]
```diff
@@ -153,7 +154,7 @@ private void deactivateThroughClient() throws Exception {
 
             log.info(">>> Try to deactivate ...");
 
-            state.active(false);
+            state.state(ClusterState.INACTIVE, true);
```
Comment (NSAmelchev) [inline modules/clients/src/test/java/org/apache/ignite/jdbc/thin/MvccJdbcTransactionFinishOnDeactivatedClusterSelfTest.java]: Can static import be used? like previuos test used

File: modules/core/src/main/java/org/apache/ignite/internal/client/impl/GridClientClusterStateImpl.java [hunk 3]
```diff
@@ -54,16 +53,6 @@ public GridClientClusterStateImpl(
...
     }
 
-    /** {@inheritDoc} */
-    @Override public void active(final boolean active) throws GridClientException {
-        state(active ? ACTIVE : INACTIVE, true);
-    }
-
-    /** {@inheritDoc} */
-    @Override public boolean active() throws GridClientException {
-        return withReconnectHandling(GridClientConnection::currentState).get();
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/internal/client/impl/GridClientClusterStateImpl.java]: `GridClientConnection::currentState` now unused. Seems can be removed too

File: modules/core/src/main/java/org/apache/ignite/internal/cluster/IgniteClusterImpl.java [hunk 4]
```diff
@@ -313,7 +315,7 @@ public IgniteClusterImpl(GridKernalContext ctx) {
         guard();
 
         try {
-            ctx.state().changeGlobalState(active, serverNodes(), false).get();
+            ctx.state().changeGlobalState(active ? ACTIVE : INACTIVE, true, serverNodes(), false, false).get();
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/internal/cluster/IgniteClusterImpl.java]: Why `forceDeactivation` should be `true` for cluster activate?

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/ActivateCommand.java [hunk 5]
```diff
@@ -49,7 +50,7 @@
         try (GridClient client = Command.startClient(cfg)) {
             GridClientClusterState state = client.state();
 
-            state.active(true);
+            state.state(ACTIVE, true);
```
Comment (NSAmelchev) [inline modules/core/src/main/java/org/apache/ignite/internal/commandline/ActivateCommand.java]: Why `forceDeactivation` should be `true` for cluster activate?

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java (hunk 1 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: > There is no cluster deactivation, `forceDeactivation` should be false.

Previous invocations were considered forced regardless of certain state. Shouldn't we do the same way?

File: modules/clients/src/test/java/org/apache/ignite/jdbc/thin/MvccJdbcTransactionFinishOnDeactivatedClusterSelfTest.java (hunk 2 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/jdbc/thin/MvccJdbcTransactionFinishOnDeactivatedClusterSelfTest.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/client/impl/GridClientClusterStateImpl.java (hunk 3 above)
Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/internal/client/impl/GridClientClusterStateImpl.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/internal/cluster/IgniteClusterImpl.java (hunk 4 above)
Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/internal/cluster/IgniteClusterImpl.java]: All previous internal invocations were considered forced. I think we should not change it where not required. Not-forced call are considered only for user commands through command.sh/Rest .

File: modules/core/src/main/java/org/apache/ignite/internal/commandline/ActivateCommand.java (hunk 5 above)
Comment (Vladsz83) [reply modules/core/src/main/java/org/apache/ignite/internal/commandline/ActivateCommand.java]: Same. Previous invocations were considered forced. I think we should not change it where not required.

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java (hunk 1 above)
Comment (NSAmelchev) [reply modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: Previous deprecated call was forced internally for both cases. I suggest do not forcing new public method call to cluster activation.

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java [hunk 6]
```diff
@@ -87,23 +89,23 @@
...
-        state.active(false);
+        state.state(INACTIVE, true);
 
         IgniteEx ig1 = grid(0);
         IgniteEx ig2 = grid(1);
 
         assertFalse(ig1.active());
         assertFalse(ig2.active());
-        assertFalse(state.active());
+        assertEquals(state.state(), INACTIVE);
```
Comment (NSAmelchev) [inline modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: Mixed up expected-actual

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java [hunk 7]
```diff
@@ -87,23 +89,23 @@
...
 
-        state.active(true);
+        state.state(ACTIVE, false);
 
         assertTrue(ig1.active());
         assertTrue(ig2.active());
-        assertTrue(state.active());
+        assertEquals(state.state(), ACTIVE);
```
Comment (NSAmelchev) [inline modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: Mixed up expected-actual

Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: Fixed.

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java (hunk 6 above)
Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: Fixed.

File: modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java [hunk 8]
```diff
@@ -87,23 +89,23 @@
     public void testActivateDeActivate() throws GridClientException {
         GridClientClusterState state = client.state();
 
-        boolean active = state.active();
+        boolean active = state.state() == ACTIVE;
```
Comment (NSAmelchev) [inline modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: `ClusterState.active(ClusterState state)` was used.

Comment (Vladsz83) [reply modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/ChangeStateCommandHandlerTest.java]: > `ClusterState.active(ClusterState state)` was used.

`GridClientClusterState.active()` was used, no? Changed to `GridClientClusterState.state()`. Isn't is ok?

**Pull request merged**